### ISSUE TITLE
Binding with [data] is buggy

### ISSUE THREAD

**Issue by kunalgrover05** [CuppaLabs/angular2-multiselect-dropdown#212]
Related to https://github.com/CuppaLabs/angular2-multiselect-dropdown/issues/208

I couldn't reproduce the issue with a trivial example, so here is a slightly complex example where the list passed to [data] is modified.
Example here: https://stackblitz.com/edit/angular2-multiselect-dropdown-bug

**Comment by vdurif** [comment]
Hi !

Experienced the same behaviour. I think that this is due to the fact that in the **ngOnChanges** of the **AngularMultiSelect** component the property **cachedItems** is not updated.

And recently a modification of the **closeDropdown()** method added this:
```
closeDropdown() {  
        if(this.searchInput){  
            this.searchInput.nativeElement.value = "";  
            this.data = [];  
            this.data = this.cachedItems;  
            this.totalHeight = this.itemHeight * this.data.length;  
            this.totalRows = this.data.length;  
            this.updateView(this.scrollTop);  
        }  
        this.filter = "";  
        this.isActive = false;  
        this.onClose.emit(false);  
    }  
```

So when you close the dropdown, **this.data** is cleared but **this.cachedItems** doesn't contain the right list of items. 

**_For example:_**

I initialize my component with a list of empty items like this:

myData = []
`<angular2-multiselect ... [data]=myData></angular2-multiselect>`
This will initilize the **cachedItems** property of **AngularMultiSelect** in the **ngOnInit** to an empty list.

Later, I call a webservice that update my property **myData**. This triggers **ngOnChanges** in **AngularMultiSelect** but do not update **cachedItems**.

Then **onClose** is triggered and **data** property of **AngularMultiSelect** is reinitialize with **cachedItems** which is empty.

See the https://github.com/CuppaLabs/angular2-multiselect-dropdown/commit/19486c90e10d8e4da73b5efd566910c7d78ed592#diff-d3d3cae12f26c239603f67b10585dfd5

**Comment by janstadt** [comment]
Im not 100% certain this is the same issue, but we have multiple multi-selects in a component that are populating the `[data]` attribute in ngOnInit via an ajax request. Once they are all loaded, you can expand one list and see the items, but then when you expand the second list, it displays "No Data Available". If you reload the page and click on them in the opposite order, it'll happen again. 

Here is an example: https://stackblitz.com/edit/angular2-multiselect-dropdown-bug-3mpwt3 

The dropdowns are populated at a different interval and once they're loaded, you can see items in the first one clicked, but after that all items are gone for the second one and subsequently the original dropdown that displayed info originally.

**Comment by CuppaLabs** [comment]
@janstadt Thank you for the bug details and example. Taken as bug. Will resolve it as soon as possible.

**Comment by CuppaLabs** [comment]
@janstadt  Fixed. Available in v2.7.0
